UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/POSAIOValidFunctionalScenarios.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Net;
+using FluentAssertions;
+using NUnit.Framework;
+using UKHO.PeriodicOutputService.API.FunctionalTests.Helpers;
+using static UKHO.PeriodicOutputService.API.FunctionalTests.Helpers.TestConfiguration;
+
+namespace UKHO.PeriodicOutputService.API.FunctionalTests.FunctionalTests
+{
+    [Category("POSAIOValidFunctionalScenarios")]
+    public class POSAIOValidFunctionalScenarios
+    {
+        private static readonly POSWebJobApiConfiguration posWebJob = new TestConfiguration().POSWebJobConfig;
+        private static readonly POSFileDetails posDetails = new TestConfiguration().posFileDetails;
+        private static readonly FSSApiConfiguration FSSAuth = new TestConfiguration().FssConfig;
+        private string DownloadedFolderPath { get; set; }
+        private string fssJwtToken;
+
+        [OneTimeSetUp]
+        public async Task Setup()
+        {
+            HttpResponseMessage apiResponse = MockHelper.ConfigureFMAio(posWebJob.MockApiBaseUrl, posWebJob.FMConfigurationValidAIOProductIdentifier);
+            apiResponse.StatusCode.Should().Be((HttpStatusCode)200);
+            await CommonHelper.RunWebJobAio();
+        }
+
+        [Test]
+        public async Task WhenICallBatchDetailsEndpointWithValidAioBatchId_ThenBatchDetailsShouldBeCorrect()
+        {
+            HttpResponseMessage apiResponse = await GetBatchDetails.GetBatchDetailsEndpoint(FSSAuth.BaseUrl, posDetails.AIOValidBatchId);
+            apiResponse.StatusCode.Should().Be((HttpStatusCode)200);
+
+            dynamic batchDetailsResponse = await apiResponse.DeserializeAsyncResponse();
+
+            GetBatchDetails.GetBatchDetailsResponseValidationForAio(batchDetailsResponse);
+        }
+
+        [Test]
+        public async Task WhenIDownloadAioExchangeSet_ThenAdditionalAioCdFilesAreGenerated()
+        {
+            DownloadedFolderPath = await FileContentHelper.DownloadAndExtractAioZip(fssJwtToken, posDetails.AioExchangeSetBatchId);
+
+            int fileCount = Directory.GetFiles(Path.Combine(DownloadedFolderPath, posDetails.AioFolderName,posDetails.InfoFolderName), "*.*", SearchOption.TopDirectoryOnly).Length;
+            Assert.IsTrue(fileCount > 0, $"File count is {fileCount} in the specified folder path.");
+        }
```

Comment:
what about unhappy path

Reply:
Hi Ray,  For the above test we are validating the INFO folder should contain at least 1 file. So if the condition doesn't match it will automatically fail. Any other unhappy path you can suggest us.
Thanks.

Reply:
Hi Ray, As the files we are downloading from FSS they are not corrupted. We don't think so files uploaded in FSS can be corrupted. Thanks